diff --git a/tools/generate_package_config.dart b/tools/generate_package_config.dart
index 01a7a0e9469..d24470e6824 100644
--- a/tools/generate_package_config.dart
+++ b/tools/generate_package_config.dart
@@ -102,12 +102,11 @@ $overrides
 String currentSDKVersion() {
   final versionContents =
       File.fromUri(repoRoot.resolve('tools/VERSION')).readAsStringSync();
-  final lines = versionContents
-      .split('\n')
-      .where((line) => !line.startsWith('#') && line.isNotEmpty);
-  final versionParts = Map.fromEntries(lines.map((line) {
-    final parts = line.split(' ');
-    return MapEntry(parts[0], parts[1]);
-  }));
+  final lines = LineSplitter.split(versionContents);
+  final versionParts = {
+    for (var line in lines)
+      if (line.isNotEmpty && !line.startsWith('#'))
+        if (line.split(' ') case [final key, final value]) key: value
+  };
   return '${versionParts['MAJOR']}.${versionParts['MINOR']}.${versionParts['PATCH']}';
 }
